testathon/testathon_data_prepare.py

```diff
@@ -0,0 +1,441 @@
+import datetime
+import os
+import subprocess
+import time
+
+import requests
+
+
+# Configuration
+
+# get api url from env, if not provided, use default
+API_URL = os.getenv('API_URL', 'https://localhost:8030/api/controller/v2')
+USERNAME = os.getenv('USERNAME', 'admin')
+PASSWORD = os.getenv('PASSWORD', 'admin')
+
+SSH_URL = os.getenv('SSH_URL', None)
+SSH_USER = os.getenv('SSH_USER', None)
+
+print(f'API_URL: {API_URL}')
+print(f'USERNAME: {USERNAME}')
+print(f'PASSWORD: {PASSWORD}')
+print(f'SSH_URL: {SSH_URL}')
+print(f'SSH_USER: {SSH_USER}')
+
+
+VERIFY_SSL = False  # Set to True if you have valid SSL certificates
+PAGE_SIZE = 100
+
+# Disable warnings for insecure SSL (if VERIFY_SSL=False)
+requests.packages.urllib3.disable_warnings(requests.packages.urllib3.exceptions.InsecureRequestWarning)
+
+
+def run(sql_script):
+    if SSH_URL is None:
+        command = ['docker', 'exec', '-i', 'tools_postgres_1', 'psql', '-U', 'awx']
+        process = subprocess.run(command, input=sql_script.encode(), capture_output=True)
+    else:
+        # Send SQL script over SSH and pipe it into `sudo awx-manage dbshell`
+        remote_command = f'echo "{sql_script}" | sudo awx-manage dbshell'
+        ssh_command = ['ssh', f'{SSH_USER}@{SSH_URL}', remote_command]
+        process = subprocess.run(ssh_command, capture_output=True)
+
+    print(process.stderr.decode())
+    return process.stdout.decode()
+
+
+def delete_job_templates():
+    # https://localhost:8030/api/controller/v2/unified_job_templates
+    url = f'{API_URL}/unified_job_templates/?type=job_template%2Cworkflow_job_template&order_by=name&page=1&page_size=100&search=mockA_test'
+    resp = requests.get(url, auth=(USERNAME, PASSWORD), verify=VERIFY_SSL)
+
+    resp.raise_for_status()
+    data = resp.json()
+
+    # count
+    count = data['count']
+    print(f'Found {count} job templates.')
+
+    for item in data['results']:
+        delete_job_template(item['id'])
+        # print(item)
+
+
+def delete_job_template(id):
+    print(f'Deleting job template {id}')
+    # https://localhost:8030/api/controller/v2/job_templates/6/
+    #
+    Failed = True
+    while True:
+        try:
+            url = f'{API_URL}/job_templates/{id}/'
+            resp = requests.delete(url, auth=(USERNAME, PASSWORD), verify=VERIFY_SSL)
+            resp.raise_for_status()
+            print(f'Deleted job template {id}')
+            if resp.status_code not in (200, 202, 204):
+                print(f'Failed to delete job template {id}: {resp.status_code} - {resp.text}')
+            else:
+                Failed = False
+                break
+        except Exception as e:
+            print(f'Failed to delete job template {id}: {e}')
+            Failed = True
+
+        if Failed:
+            print(f'Failed to delete job template {id}, maybe jobs are still running, trying again in 10 seconds')
+            time.sleep(10)
+        else:
+            break
```

Comment:
```suggestion
    failed = True
    while True:
        try:
            url = f'{API_URL}/job_templates/{id}/'
            resp = requests.delete(url, auth=(USERNAME, PASSWORD), verify=VERIFY_SSL)
            resp.raise_for_status()
            print(f'Deleted job template {id}')
            if resp.status_code not in (200, 202, 204):
                print(f'Failed to delete job template {id}: {resp.status_code} - {resp.text}')
            else:
                failed = False
                break
        except Exception as e:
            print(f'Failed to delete job template {id}: {e}')
            failed = True

        if failed:
            print(f'Failed to delete job template {id}, maybe jobs are still running, trying again in 10 seconds')
            time.sleep(10)
        else:
            break
```

only Classnames should start with a capital

Reply:
Oooh, good catch!